### PR TITLE
migrations: migrate to flask-migrate (bug 1694242)

### PR DESCRIPTION
**Pull request by zzzeid**
See note below before landing and deploying. This deployment will need to be coordinated with the Cloud Ops team, and needs to be tested on the development environment with a fresh clone of the production DB before we deploy to production.

- remove flask-alembic dependency
- add flask-migrate dependency
- update code to work with flask-migrate
- upgrade alembic, sqlalchemy packages
- remove all previous migrations
- reinitialize all migrations based on latest state
- remove init cli command (keep init_s3 for now)
- update tasks
- update readme with instructions to generate migrations

NOTE:

This PR will need to be deployed manually, and the `alembic_version` table needs to be updated with `version_num` set to `7883d80258fb` manually, before the migration step is performed.

### CODE REVIEW

File: migrations/versions/c02a75d4f418_.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
+"""empty message
```
Comment (zzzeid) [inline migrations/versions/c02a75d4f418_.py:1]: Should replace this with a message describing this initial migration as matching the previous state of the database.

File: landoapi/cli.py [hunk 2]
```diff
@@ -52,24 +52,16 @@ def cli():
...
-            aws_secret_key=os.environ["AWS_SECRET_KEY"],
-            endpoint_url=os.environ["S3_ENDPOINT_URL"],
-        )
-        s3.create_bucket(Bucket=os.environ["PATCH_BUCKET_NAME"])
+    s3 = patches.create_s3(
+        aws_access_key=os.environ["AWS_ACCESS_KEY"],
+        aws_secret_key=os.environ["AWS_SECRET_KEY"],
+        endpoint_url=os.environ["S3_ENDPOINT_URL"],
+    )
+    s3.create_bucket(Bucket=os.environ["PATCH_BUCKET_NAME"])
```
Comment (zzzeid) [inline landoapi/cli.py:64]: Maybe there's a better place for this elsewhere?

File: migrations/alembic.ini [hunk 3]
```diff
@@ -0,0 +1,50 @@
+# A generic, single database configuration.
```
Comment (zzzeid) [inline migrations/alembic.ini:1]: Should verify that the settings in this file are being read correctly in all environments.

File: landoapi/cli.py [hunk 4]
```diff
@@ -52,24 +52,16 @@ def cli():
...
-def init(init_s3):
-    """Initialize Lando API (Create the DB, etc.)"""
-    # Create the database and set the alembic version to
-    # head revision.
-    from landoapi.storage import alembic, db
-
-    db.create_all()
-    alembic.stamp("head")
-
+def init_s3(init_s3):
```
Comment (cgsheeh) [inline landoapi/cli.py:55]: If we keep this function you'll need to remove the `init_s3` argument for it to work.

File: landoapi/cli.py (hunk 2 above)
Comment (cgsheeh) [reply landoapi/cli.py:64]: If you can think of somewhere else to put it that would be r+ from me. Though IIRC you have some work upcoming to remove S3 from Lando entirely, I would be fine with leaving this here until that lands and then this becomes redundant.

File: migrations/alembic.ini (hunk 3 above)
Comment (cgsheeh) [inline migrations/alembic.ini:1]: Needs a license header.

File: migrations/env.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
+from __future__ import with_statement
```
Comment (cgsheeh) [inline migrations/env.py:1]: License.

File: migrations/versions/c02a75d4f418_.py (hunk 1 above)
Comment (cgsheeh) [inline migrations/versions/c02a75d4f418_.py:1]: License.

**Review by cgsheeh — CHANGES_REQUESTED** [review]
Just need to add license headers then this is r+. Thanks for doing this, it was a huge headache when working on autoformat.

File: migrations/versions/c02a75d4f418_.py (hunk 1 above)
Comment (zzzeid) [reply migrations/versions/c02a75d4f418_.py:1]: We don't normally put license headers on these autogenerated migration files (check all deleted migration files below/above).

**Review by cgsheeh — APPROVED** [review]

**Pull request closed without merging**